packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts

```diff
@@ -305,6 +305,349 @@ describe("ModelsPropertyOptional Rest Client", () => {
+    }
+  });
+
+  it("should put default union string literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/string/literal/default")
+        .put({
+          body: {}
+        });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get all union int literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/int/literal/all")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body.property, 2);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get default union int literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/int/literal/default")
+        .get();
+      assert.strictEqual(result.status, "200");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put all union int literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/int/literal/all")
+        .put({
+          body: { property: 2 }
+        });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put default union int literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/int/literal/default")
+        .put({
+          body: {}
+        });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get all union float literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/float/literal/all")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body.property, 2.3);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get default union float literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/float/literal/default")
+        .get();
+      assert.strictEqual(result.status, "200");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put all union float literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/float/literal/all")
+        .put({
+          body: { property: 2.3 }
+        });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put default union float literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/float/literal/default")
+        .put({
+          body: {}
```

Comment:
why this is empty?

Reply:
The default value is set to empty like the other cases.